lib/Doctrine/DBAL/Driver/PDOStatement.php

```diff
@@ -92,15 +92,9 @@ public function bindParam($column, &$variable, $type = ParameterType::STRING, $l
     /**
      * {@inheritdoc}
      */
-    public function closeCursor()
+    public function closeCursor() : void
     {
-        try {
-            return $this->stmt->closeCursor();
-        } catch (\PDOException $exception) {
```

Comment:
Does this mean it can now throw generic PDOException here? Should we wrap it?

Reply:
I thought I'd put it into the description but I hadn't. `PDOStatement::closeCursor()` doesn't throw exceptions in any of the scenarios we have in our tests including the newly added ones. That's why the interface annotation doesn't declare any `@throws`. I cannot imagine a situation when the result cannot be freed on a valid statement and how this could be handled.

AFAIK, procedural APIs like `oci_cancel($stmt)` and the similar from other drivers only return `false` when the statement handle is invalid which is impossible in our case.